### PR TITLE
Feature/check 4 3

### PR DESCRIPTION
**Pull request by badamowicz**
Hi, all!
As announced in [this PR](https://github.com/containers/podman-security-bench/pull/6),
here comes the implementation of check\_4\_3. Let me explain a few things why I did it this way.

## Whitelist vs. Blacklist

In the beginning I thought it was a good idea to also use a whitelist for 4.3, but it wasn't at all.
Different distributions will provide different package managers and those in turn will provide
different formats of the output of the package list. This rendered it almost impossible to provide a
reasonable whitelist.
So I changed my mind and also intruduced blacklist which made it pretty easy to achieve the goal.

## New directory structure

I removed the former _whitelist_ directory and an originally introduced _blacklist_ directory and
instead now only use a single directory named _default-lists_ which contains both blacklists and
whitelists. Not a big issue, but makes the handling of the files a bit easier.

## New function

I've introduced a new function `get_list_cmd()` which provides the command for listing packages for
their respective distribution type. Currently it looks like this:

```bash
get_list_cmd() {
  distro="${1}"

  case "$distro" in
    fedora,rhel)
      echo -n "rpm -qa"
      return
      ;;
    debian|ubuntu)
      echo -n "apt list --installed"
      return
      ;;
    *)
      echo -n "rpm -qa"
      return
      ;;
  esac
}
```

There is obviously more room left for other distribution types as you can see. So feel free to
contribute more types if you thinks that's necessary. I wasn't able to evaluate all available distros in
my local environment.

Looking forward to your suggestions!
Bernd

### CODE REVIEW

**Comment by badamowicz** [comment]
Encountered a bug in check 5_8 if the whitelist is empty. Don't merge so far. Will provide a fix.

**Comment by badamowicz** [comment]
Bug is fixed in [6f3dcce](https://github.com/containers/podman-security-bench/pull/7/commits/6f3dcce8c5ca552327ad5291eabcab6610ec1ea4). It actually only occured if the lists of ports inside the container was empty.

File: functions/helper_lib.sh [hunk 1]
```diff
@@ -149,3 +149,25 @@ yell "# ------------------------------------------------------------------------
 # Based on the CIS Docker Benchmark 1.3.1.
 # --------------------------------------------------------------------------------------------"
 }
+
+# retrieve command for package manager
+get_list_cmd() {
+  distro="${1}"
+
+  case "$distro" in
+    fedora,rhel)
```
Comment (rhatdan) [inline functions/helper_lib.sh]: Add centos?

File: podman-security-bench.sh [hunk 2]
```diff
@@ -20,6 +20,8 @@ myname=$(basename "${this_path%.*}")     ## file name of this script.
 
 # default path to whitelist files
 wl_path=whitelists
```
Comment (rhatdan) [inline podman-security-bench.sh]: Can we change these to the more cultural correct names allowlist and denylist.

File: functions/helper_lib.sh (hunk 1 above)
Comment (tumauwt) [reply functions/helper_lib.sh]: rocky as well?

Comment (rhatdan) [reply functions/helper_lib.sh]: sure

Comment (badamowicz) [reply functions/helper_lib.sh]: Added both. Please check if it's correct. Wasn't sure.

File: podman-security-bench.sh (hunk 2 above)
Comment (badamowicz) [reply podman-security-bench.sh]: Already changed. However I don't think that the colors black or white mean an offense to anybody.

**Comment by badamowicz** [comment]
Converted this to a draft because after internal discussions about check 4.3 we think it might be a better solution to use a whitelist instead of a blacklist. So I will have to re-think the implementation.

File: podman-security-bench.sh (hunk 2 above)
Comment (rhatdan) [reply podman-security-bench.sh]: Maybe, Maybe not, but we have been told to use allow and deny versus Black and White.

**Comment by badamowicz** [comment]
Completely re-implemented 4.3. No more whitelists/blacklists. Instead only allow files.

**Comment by rhatdan** [comment]
Nice work, can you squash your commits and rewrite the commit message on github.

**Comment by badamowicz** [comment]
> Nice work, can you squash your commits and rewrite the commit message on github.

Thanks! All squashed to a single commit now.

**Pull request merged**